### ISSUE TITLE
Honor OMVS umask when cloning a repository

### ISSUE THREAD

**Issue by gngrossi** [zigi/zigi#479]
ZIGI v3r18
After cloning, the OMVS directories and files have file permissions that should be more restrictive.
A common umask of 022 will result in directory permissions 755 and file permissions 644.

--

bash-5.2$ umask
0022

/u/@02858/Projects/Repos
bash-5.2$ ls -l
drwxrwxrwx  10 @02858   @ISCICS1    8192 Aug 29 09:22 pdsegen

/u/@02858/Projects/Repos/pdsegen
bash-5.2$ ls -l
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.ASM
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.EXEC
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.LOAD
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.PANELS
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.PDS
drwxrwxrwx   2 @02858   @ISCICS1    8192 Aug 29 09:22 PDSEGEN.SAMPLE.BACKUP
-rw-rw-rw-   1 @02858   @ISCICS1   20384 Aug 29 09:22 README.md
-rw-rw-rw-   1 @02858   @ISCICS1    1079 Aug 29 09:22 zginstall.readme
-rwxrwxrwx   1 @02858   @ISCICS1   74324 Aug 29 09:22 zginstall.rex

**Comment by lbdyck** [comment]
On my z/OS system in /etc/profile I found this for me:

![image](https://github.com/zigi/zigi/assets/42328411/548d7547-3f50-4f19-96c9-13caa9b870b7)

does this help?

**Comment by gngrossi** [comment]
Our default umask is the same in /etc/profile.  Each user can set their own umask in their .profile. This request is to use what's active for the user. The umask command will display it.  Why is 777 and 666 being used?
Does this help?
thanks

**Comment by lbdyck** [comment]
I don't know - that will require some research into zigi and how it does the copy.  I don't see any chmod's in the code, nor do I see any umask.

**Comment by lbdyck** [comment]
note this will not affect any current repos unfortunately but will correct things for newly created repos

**Comment by gngrossi** [comment]
ZIGI v3r19
After cloning, the OMVS directories and files have file permissions that match umask 022.

RC=(0) [SYSA] bash-5.2$ umask
0022

RC=(0) [SYSA] bash-5.2$ cd Repos
RC=(0) [SYSA] bash-5.2$ ls -l
drwxr-xr-x  10 @02858   @ISCICS1    8192 Aug 30 17:22 pdsegen

RC=(0) [SYSA] bash-5.2$ cd pds*;ls -l
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.ASM
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.EXEC
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.LOAD
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.PANELS
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.PDS
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 30 17:22 PDSEGEN.SAMPLE.BACKUP
-rw-r--r--   1 @02858   @ISCICS1   20384 Aug 30 17:22 README.md
-rw-r--r--   1 @02858   @ISCICS1    1079 Aug 30 17:22 zginstall.readme
-rwxr-xr-x   1 @02858   @ISCICS1   74324 Aug 30 17:22 zginstall.rex

--

After dynamically changing the umask (umask 027 command), deleting the repo and cloning, the directory and file permissions did not reflect umask 027 (they were the same as umask 022).

Does the umask command need to be in the $HOME/.profile ?

thanks

**Comment by lbdyck** [comment]
Existing repos won't be affected as they have their permissions. Only newly created repos will be affected. 

Let me know

**Comment by gngrossi** [comment]
After using umask 022 successfully, I dynamically changed to umask 027 and deleted the repo and re-cloned. I will try it again.
thanks

**Comment by lbdyck** [comment]
I believe that the permissions that are already in the repo will be honored - thus the change to support umask would only work for newly created repositories and not for existing ones. I could be wrong.

**Comment by lbdyck** [comment]
I was wrong - the permissions after a clean clone fit the umask.

Thank you for reporting, and testing, this issue and fix.

**Comment by gngrossi** [comment]
ZIGI v3r19

After dynamically changing the umask to 027, deleting the repo (datasets/files) and re-cloning, the directory and file permissions did not reflect umask 027.

--

Before cloning:
RC=(0) [SYSA] bash-5.2$ cd Repos
RC=(0) [SYSA] bash-5.2$ ls -l
drwxr-x---   5 @02858   @ISCICS1    8192 Dec  7  2022 Fed_JCL

RC=(0) [SYSA] bash-5.2$ umask 027
RC=(0) [SYSA] bash-5.2$ umask
0027

<Cloned the repo>

                      ZIGI Local Repository Summary Info 
  Name          :  pdsegen                               
  Repository    :  pdsegen                               
  Branch        :  master                                
  Remote        :  git@github.com:lbdyck/pdsegen.git     
                                                         
  Home Directory:  /u/@02858/Projects/Repos/pdsegen      
  Category      :  Tools                                 
  z/OS HLQ      :  @02858.ZIGI2                          
  Ignore Qual   :  2                                     
  Last Access   :  31 Aug 2023                           
  Default Userid:                                        
  Default Push  :  N                                     
  Read Only     :  No                                    


After cloning:
With umask 027, directories would be 750 and files would be 640.

RC=(0) [SYSA] bash-5.2$ ls -l
drwxr-x---   5 @02858   @ISCICS1    8192 Dec  7  2022 Fed_JCL
drwxr-xr-x  10 @02858   @ISCICS1    8192 Aug 31 15:21 pdsegen

RC=(0) [SYSA] bash-5.2$ ls -l pds*
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.ASM
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.EXEC
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.LOAD
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.PANELS
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.PDS
drwxr-xr-x   2 @02858   @ISCICS1    8192 Aug 31 15:21 PDSEGEN.SAMPLE.BACKUP
-rw-r--r--   1 @02858   @ISCICS1   20384 Aug 31 15:21 README.md
-rw-r--r--   1 @02858   @ISCICS1    1079 Aug 31 15:21 zginstall.readme
-rwxr-xr-x   1 @02858   @ISCICS1   74324 Aug 31 15:21 zginstall.rex


thanks

**Comment by lbdyck** [comment]
It worked for me with umask 0027 - what am I missing?

![image](https://github.com/zigi/zigi/assets/42328411/f061c39b-d8d6-41f8-a78e-0e6e9fddcfdc)

**Comment by gngrossi** [comment]
My mistake. I needed to update my .profile with umask 027 and it works as expected. Sorry about that.

Is it possible to include a umask field on the Clone Repository panel?  It would be used to override the /etc/profile or .profile.

thanks

**Comment by lbdyck** [comment]
Thanks for the confirmation.

The issue is more than just when Cloning - it is also when Creating and perhaps when doing a Branch (don't know about that one).

I'll discuss with Henri.

**Comment by gngrossi** [comment]
So I understand, is the umask value in /etc/profile used if a user doesn't have a umask in their .profile?
thanks

**Comment by lbdyck** [comment]
Yes - the active umask from /etc/profile or ~/.profile will be used.

After some thought I think we will leave this as is so that zigi works the same if you used the `git` command from an OMVS prompt.  Otherwise it could be confusing.

Make sense?

**Comment by gngrossi** [comment]
OK...
If you have an active shell session with either with OMVS or an SSH client, issue a umask xxx command and then enter a git clone command, the dynamic umask change will be in effect, right?  For me, I use umask 022 as a default but will dynamically use umask 027 when needed without changing my .profile.

**Comment by lbdyck** [comment]
understand - i'll let Henri chime in - adding it to the zigi dialog isn't difficult.

**Comment by lbdyck** [comment]
Should this be at the repository level or apply to all repositories?

**Comment by gngrossi** [comment]
Looks to be working. Tested umask 022 and umask 027.
After I updated my $HOME/.profile, I needed to exit ZIGI for the umask to be honored.
Is this expected?
thanks

**Comment by MN-Mike** [comment]
I cannot speak to the specific change but UMASK is only persisted for the life of the session.  So it does make sense that you would need to establish a new session (which is when the $HOME/.profile is executed) to see any change you make to the .profile

**Comment by gngrossi** [comment]
My first test was using my default umask 022. I logged into TSO/ISPF, started ZIGI and then cloned a repo.
The zfs repo directories on file system were proper.

Next, I updated my .profile using umask 027. From ZIGI, (without exiting), I deleted (removed all files) and cloned the repo.
The zfs repo directories on the file system did not reflect umask 027.
It wasn't until I exited ZIGI and started it again, that the new umask was honored.

Hope that explains it.
